src/hb-open-file-private.hh

```diff
@@ -295,11 +295,13 @@ struct OpenTypeFontFile
 {
   static const hb_tag_t tableTag	= HB_TAG ('_','_','_','_'); /* Sanitizer needs this. */
 
-  static const hb_tag_t CFFTag		= HB_TAG ('O','T','T','O'); /* OpenType with Postscript outlines */
-  static const hb_tag_t TrueTypeTag	= HB_TAG ( 0 , 1 , 0 , 0 ); /* OpenType with TrueType outlines */
-  static const hb_tag_t TTCTag		= HB_TAG ('t','t','c','f'); /* TrueType Collection */
-  static const hb_tag_t TrueTag		= HB_TAG ('t','r','u','e'); /* Obsolete Apple TrueType */
-  static const hb_tag_t Typ1Tag		= HB_TAG ('t','y','p','1'); /* Obsolete Apple Type1 font in SFNT container */
+  enum {
+    CFFTag		= HB_TAG ('O','T','T','O'), /* OpenType with Postscript outlines */
+    TrueTypeTag	= HB_TAG ( 0 , 1 , 0 , 0 ), /* OpenType with TrueType outlines */
+    TTCTag		= HB_TAG ('t','t','c','f'), /* TrueType Collection */
+    TrueTag		= HB_TAG ('t','r','u','e'), /* Obsolete Apple TrueType */
+    Typ1Tag		= HB_TAG ('t','y','p','1')  /* Obsolete Apple Type1 font in SFNT container */
+  };
```

Comment:
This was needed otherwise linker had this issue,

```
  CXXLD    libharfbuzz-subset.la
Undefined symbols for architecture x86_64:
  "OT::OpenTypeFontFile::TrueTypeTag", referenced from:
      _hb_subset_face_reference_table(hb_face_t*, unsigned int, void*)in libharfbuzz_subset_la-hb-subset.o
  "OT::OpenTypeFontFile::CFFTag", referenced from:
      _hb_subset_face_reference_table(hb_face_t*, unsigned int, void*)in libharfbuzz_subset_la-hb-subset.o
ld: symbol(s) not found for architecture x86_64
```

Reply:
That's some really stupid compiler...

I'm not sure what to feel about this change. We have many other places that we define `static const hb_tag_t` values. This is now inconsistent.

Do we care about that particular compiler? Honest question.

Reply:
I’d say we shouldn’t, it is an ancient compiler for an ancient, unsupported by its vendor, platform. Downstream can maintain there own patch-set if they want, it is not like HarfBuzz is the only package that will break on it.

Reply:
The fact,

* It provides useful insights and it just needs some trivial changes which some are considered nice,
* There are people complaining about it, #853 and https://github.com/macports/macports-ports/pull/1466
* We can have it on our CI and we can have SSH access to the bot when you don't know what is wrong

makes it eligible for short term support. If you guys don't agree, at least lets have one release with it, write some compiler policy somewhere about it and leave the support those don't fit (removing their bots).

Reply:
For example see https://bugs.chromium.org/p/chromium/issues/detail?id=820422 and https://groups.google.com/a/chromium.org/forum/#!topic/cxx/N7weI2XylMo

> I assume we would move to a community-supported model for MSVC in Chromium's code – no bot coverage, but patches to keep things building are accepted – just like our current gcc setup.

Even on not supported compilers, it would be nice to accept needed patches at least.

And from, http://blog.llvm.org/2018/03/clang-is-now-used-to-build-chrome-for.html

> Building your project with several compilers is generally considered good for code health, but in Chrome we found that Clang’s diagnostics found most problems and we were mostly battling compiler bugs (and if another compiler has a great new diagnostic, we can add that to Clang).

If we follow Chromium lead here [sarcastically], we can drop both MSVC and GCC support and bots :)

Reply:
I understand the value of different compilers and bots. As long as we know why we are doing it and the workaround are not ugly. In this case, we clearly are working around a compiler bug.

I just miss the days that all our commits were bug fixes or feature implementations. :) Guess that's the price of popularity... :D